lib/server.js

```diff
@@ -524,10 +522,13 @@ function createServer(options) {
                 var size_histogram = collector.getCollector(
                     METRIC_SIZE_HISTOGRAM);
 
-                var labels = { type: query.type() };
-                query_counter.increment(labels);
-                latency_histogram.observe(lat / 1000, labels);
-                size_histogram.observe(query.bytesSent, labels);
+                var queryType = query.type();
+                if (queryType) {
+                        var labels = { type: query.type() };
```

Comment:
Why have queryType if it is only going to be used once?  Did you mean to reuse it on line 527?

Reply:
Indeed, modifying in a bit.